### PR TITLE
AzureMonitor: Remove unused editor component

### LINKED ISSUE #18057: Azure Monitor: Refactor to use common QueryField component instead of custom one
Currently the azure monitor datasource makes use of `public/app/plugins/datasource/grafana-azure-monitor-datasource/editor/query_field.tsx` rather than `public/app/features/explore/QueryField.tsx`. It would be nice to refactor so that the previous file isn't needed anymore.

### PR DESCRIPTION
**Pull request by iwysiu**
**What this PR does / why we need it**:
Removes the old, unused query editor from the Azure Monitor code.

**Which issue(s) this PR fixes**:

Fixes #18057

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/query_ctrl.ts [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import { defaultsDeep, find, map, isString } from 'lodash';
 import { QueryCtrl } from 'app/plugins/sdk';
 import TimegrainConverter from './time_grain_converter';
-import './editor/editor_component';
```
Comment (sunker) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/query_ctrl.ts:4]: Is this file used? I believe you should be able to remove this and the annotations_query_ctrl.

**Review by andresmgot — APPROVED** [review]
LGTM!

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/query_ctrl.ts (hunk 1 above)
Comment (joshhunt) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/query_ctrl.ts:4]: Yup - feel free to remove query_ctrl and annotations_query_ctrl also @iwysiu !

Comment (sunker) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/query_ctrl.ts:4]: And the [partials](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/prometheus/partials/) too. :)

**Pull request merged**